### DISCUSSION TITLE
RuntimeError: Wifi Unknown Error 0x1101 with Flash Encryption

### DISCUSSION THREAD

**Discussion by amplatzer** [micropython/micropython#14033] [ESP32]
MPY v1.22.0-dirty on 2024-03-05
IDF  v5.1.1 
Chip:  esp32-pico-d4

The code works just fine, but with **Flash Encryption** enabled, it fails to run. any idea why?

I (780) boot: Loaded app from partition at offset 0x20000
I (780) boot: Checking flash encryption...
I (780) flash_encrypt: flash encryption is enabled (3 plaintext flashes left)
I (787) boot: Disabling RNG early entropy source...

icroPython v1.22.0-dirty on 2024-03-05; Generic ESP32 module with ESP32
Type "help()" for more information.
>>> import network
>>> station = network.WLAN(network.STA_IF)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
RuntimeError: Wifi Unknown Error 0x1101

**Comment by jimmo** [comment]
@amplatzer I don't know sorry, but perhaps https://github.com/micropython/micropython/pull/12968 (and the original issue here https://github.com/micropython/micropython/issues/12750) might give some hints?

**Reply by amplatzer** [reply]
Thanks, Jimmo, I saw those posts, but it was a different Error(0x0101), And I thought it was fixed and merged, so I tried v1.22.0. Should I test with latest mater build?

**Reply by amplatzer** [reply]
And none of them mention Flash Encryption. So I'm not sure.

**Comment by drc-ksafin** [comment]
Hey @amplatzer, any chance you figured this out? Having the exact same issue right now. If you do have any tips, it'd be super appreciated.

**Reply by glenn20** [reply]
The espressif wifi drivers use the NVS internally for storing information. 

It may be that special configuration is required in the espressif wifi drivers to support an encrypted NVS partition. This is speculation, but may help point to a solution.
